fetchai/agents-aea
# [AEA-481 AEA-484 AEA-485 AEA-550] Feature - add p2p noise connection

## Changes

Add a P2P Noise Connection that implement an AEA Connection that deploys an noise node (go code/binary) and exchanges envelopes through named pipes.

## Missing/TOFIXs
- To do

## How to test
1. `git fetch && git checkout feature/add-p2p-noise`
2. `python3 -m pip install -e .[all]`
3. [`aea init &&`]`aea create my_agent && cd my_agent`
4. `aea add --local connection fetchai/p2p_noise:0.1.0`
5. `aea remove connection fetchai/stub:0.2.0`
6. `aea config set agent.default_connection fetchai/p2p_noise:0.1.0`
7. `aea install`
8. `aea run`

- Noise node logs are saved to "noise_node.log" file by default

Note: 
- for code history: https://github.com/fetchai/agents-p2p/tree/251ffaacafe4eba264df093d19d3357c1c852dc0/pocs/p2p_noise_pipe
- for pr history check https://github.com/fetchai/agents-aea/pull/1075
